python/cm/upgrade.py

```diff
@@ -256,6 +256,23 @@ def rpm_cmp_reverse(obj1, obj2):
         return res
 
 
+def re_count_issue(obj, upgrade):
+    """Re-count of issues from the bottom up"""
+    if obj.prototype.type == 'cluster':
+        for proto in Prototype.objects.filter(bundle=upgrade.bundle, type='service'):
+            for co in ClusterObject.objects.filter(cluster=obj, prototype=proto):
+                for comp in Prototype.objects.filter(parent=proto, type='component'):
```

Comment:
There is no needs to iterate over component's prototype. Filter from ServiceComponent.objects.filter(                            cluster=obj, service=co) is enough to got a component. 

In current realisation for every comp you will have only one sc and no way to got two or more sc in line 265. So, just remove loop from line 264

Reply:
Fixed